### ISSUE TITLE
Redirect to client after enrollment

### ISSUE THREAD

**Issue by AminSaqi** [goauthentik/authentik#1203]
**Describe your question/**
I want to redirect user to client after successful enrollment, just like what the authorization flow does. 

I imported `Enrollment with email verification` flow example, and it have `default-enrollment-user-login` stage at the end of stage binding. However, it redirects the user to the `Library` page of the authentik panel, but I want it to redirect the user to the client.

What should I do?

Thanks in advance.

**Version and Deployment (please complete the following information):**
 - authentik version: 2021.7.1
 - Deployment: docker-compose

**Comment by BeryJu** [comment]
Hi, you can use an expression policy to write to the flow context and set `redirect` to the URL you want to redirect:

```python
context["redirect"] = "http://foo.bar.baz"
return True
```

and then bind that policy to the `user_login` stage-binding, and make sure to enable the `re_evaluate_policies` flag.

**Comment by AminSaqi** [comment]
Thank you, but got error:

Whoops! 
Something went wrong! Please try again later. 

My flow diagram:
![pol1](https://user-images.githubusercontent.com/39796124/127770891-903ba06c-6659-460f-a8c0-f975c04f07f8.jpg)

My expression policy:
![pol2](https://user-images.githubusercontent.com/39796124/127770905-b26d1a2d-74ca-4304-a10b-eda682db2906.jpg)

Stage binding:
![pol3](https://user-images.githubusercontent.com/39796124/127770915-56b0356b-bfbc-41d0-8bfa-6ca51c0fc14b.jpg)

Anything I miss?

**Comment by BeryJu** [comment]
Can you check if there are any logs in your authentik System Log under Events -> Log?

**Comment by AminSaqi** [comment]
Yes there are:

![logs](https://user-images.githubusercontent.com/39796124/127771127-6ba86d6f-b3e6-4de0-a5d8-8211328e7097.jpg)

Here is the last policy execution:

![exec](https://user-images.githubusercontent.com/39796124/127771217-5e8c3b8d-5abc-41d4-9baf-24b727de3181.jpg)

**Comment by BeryJu** [comment]
Hmm weird, normally when the "Something went wrong" message is shown, a `system_exception` event is created with a full stacktrace. Can you look in the container logs if theres any error in there?

From the events you posted above everything looks right.

**Comment by BeryJu** [comment]
Actually I just noticed that unhandled exceptions during flow executions are not logged as system exception events, that'll be fixed. So in your case there should definitely be an error in the docker logs.

**Comment by AminSaqi** [comment]
I think these logs are relevant to the flow:

```
{"event": "/api/v2beta/flows/executor/register/?query=token%3D37a4703372a74a509d7da3f205be162f", "host": "<myIp>", "level": "info", "logger": "authentik.asgi", "method": "GET", "pid": 21, "request_id": "961703fb9726459588b778f6ec7ab14a", "runtime": 0, "scheme": "https", "size": 0.031, "status": 302, "timestamp": "2021-08-01T12:35:15.137369"}

{"event": "NoReverseMatch(\"'https' is not a registered namespace\")", "flow_slug": "register", "host": "sso.quantct.com", "level": "warning", "logger": "authentik.flows.views", "pid": 21, "request_id": "b3e5a18b726b476d8759cfa7958afa25", "timestamp": "2021-08-01T12:35:15.514751"}

{"event": "/api/v2beta/flows/executor/register/?query=token%3D37a4703372a74a509d7da3f205be162f", "host": "<myIp>", "level": "info", "logger": "authentik.asgi", "method": "GET", "pid": 21, "request_id": "b3e5a18b726b476d8759cfa7958afa25", "runtime": 427, "scheme": "https", "size": 0.103, "status": 200, "timestamp": "2021-08-01T12:35:15.531975"}

{"event": "/api/v2beta/events/events/?ordering=-created&page=1&page_size=20&search=", "host": "<myIp>", "level": "info", "logger": "authentik.asgi", "method": "GET", "pid": 18, "request_id": "52b3f5a0cad94d4296c70c9d588decbc", "runtime": 33728, "scheme": "https", "size": 0.028, "status": 200, "timestamp": "2021-08-01T12:35:23.568441"}

{"event": "/api/v2beta/events/notifications/?ordering=-created&page_size=1&seen=false", "host": "<myIp>", "level": "info", "logger": "authentik.asgi", "method": "GET", "pid": 18, "request_id": "e170153c0f2143e889f2d186630dd604", "runtime": 119, "scheme": "https", "size": 0.019, "status": 200, "timestamp": "2021-08-01T12:35:23.645791"}
```

I only see one warning - the second one.

However, yes, previously I saw several errors in docker container logs, but no one in the logs page. I also saw several error logs on the logs page, too.

**Comment by AminSaqi** [comment]
Actually, I saw some exception notification (NOT relevant to this thread) in the Notifications side-bar, not in logs page.

**Comment by BeryJu** [comment]
Found the issue, as a security measure you can only redirect to either authentik Views or URLs relative to authentik, but I'll change it that when setting `redirect` from within a flow you can redirect wherever. Fix will be in 2021.7.2
